src/org/labkey/test/components/domain/DomainFieldRow.java

```diff
@@ -600,6 +602,164 @@ public DomainFieldRow clickRemoveOntologyConcept()
+    {
+        getWrapper().setFormElement(Locator.tagWithName("input", "value").findElement(this), updatedValue);
+        return this;
+    }
+
+    /**
+     * Update the selected value in the TextChoice field. This will select the given field.
+     *
+     * @param originalValue The original value to select and change.
+     * @param newValue The new value to replace it.
+     * @return This field row.
+     */
+    public DomainFieldRow updateTextChoiceValue(String originalValue, String newValue)
+    {
+        selectTextChoiceValue(originalValue);
+        setUpdateTextChoiceValue(newValue);
+        WebDriverWrapper.waitFor(this::isTextChoiceApplyButtonEnabled, "'Apply' button is not enabled.", 1_000);
+        Locator.button("Apply").findElement(this).click();
+
+        return this;
+    }
+
+    /**
+     * Update a TextChoice value and expect an error. Can happen if the new value already exists in the list of values.
+     *
+     * @param originalValue The original value to select and change.
+     * @param newValue The new value to replace it.
+     * @return The error message.
+     */
+    public String updateTextChoiceValueExpectError(String originalValue, String newValue)
+    {
+        updateTextChoiceValue(originalValue, newValue);
+        WebElement alert = BootstrapLocators.errorBanner.waitForElement(this, 1_000);
+        return alert.getText();
+    }
+
+    /**
+     * Update a TextChoice value that is currently being used and expect an info message with the number updated.
+     *
+     * @param originalValue The original value to select and change.
+     * @param newValue The new value to replace it.
+     * @return The info message indicating the number of entites updated.
+     */
+    public String updateLockedTextChoiceValue(String originalValue, String newValue)
+    {
+        updateTextChoiceValue(originalValue, newValue);
+        WebElement alert = BootstrapLocators.infoBanner.waitForElement(this, 1_000);
+        return alert.getText();
+    }
+
+    /**
+     * Check to see if the 'Delete' button is enabled for a TextChoice field. Note: The button is only visible if a
+     * value is selected.
+     *
+     * @return True if the button is enabled, false otherwise.
+     */
+    public boolean isTextChoiceDeleteButtonEnabled()
+    {
+        WebElement button = Locator.button("Delete").findElement(this);
+        return button.isEnabled();
+    }
+
+    /**
+     * Delete a value from the list of values for a TextChoice field. This will select the value before deleting it.
+     *
+     * @param value The value to delete.
+     * @return This field row.
+     */
+    public DomainFieldRow deleteTextChoiceValue(String value)
+    {
+        selectTextChoiceValue(value);
+        WebElement button = Locator.button("Delete").findElement(this);
+        WebDriverWrapper.waitFor(button::isEnabled, "'Delete' button is not enabled.", 1_000);
+        button.click();
+        return this;
+    }
```

Comment:
Should be instantaneous, but to be safe I added a wait for the delete button to disappear (won't be disabled because no value will be selected after the selected value is deleted).